docs/doxygen/events.md

```diff
@@ -0,0 +1,69 @@
+### Execution Engine and Events
+
+The execution engine is an execution context that supports event-driven network
+execution on the CUDA streams, CPU threads, and DPU threads. It is intended to
+interact with execution threads that are asynchronous (offloaded collective
+execution) which can be implemented on GPUs, DPUs, or remote CPUs.  
+
+In the event-driven execution, the events are generated when there is an event
+in the UCC library. The event can include that an operation posted or completed.
+The users can launch the UCC collective operations through the triggered
+operations. Besides the execution engine, events and event queues are key for
+event-driven execution. The operations on the execution engines generate events
+that are stored on the event queues. The valid events are defined by @ref
+ucc\_event\_type\_t and valid event queues are defined by @ref
+ucc\_event\_queue\_type\_t.
+
+The interaction between the user and library is through the UCC interfaces. @ref
+ucc\_ee\_create creates execution engines. The number and type of events queues
+on the execution engine are configurable during the creation time. The user or
+library can generate an event and post it to the event queues using @ref
+ucc\_ee\_set\_event interface. The user can poll or wait on the event queue with
+@ref ucc\_ee\_poll and @ref ucc\_ee\_wait interfaces respectively.
```

Comment:
need remove event queues references from above text

Reply:
Done. I still would like to bring out - event queues. Probably, as advice to implementations.